File: server-side/src/main/java/com/guegue/service/PropertyService.java
package com.guegue.service;

import com.guegue.model.Property;
import org.springframework.stereotype.Service;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;
import java.security.Principal;
import java.util.List;

public interface PropertyService {
    Property getPropertyByID(int id, Principal principal);

    List<Property> getProperties(Principal principal);

    Property updateProperty(Property updatedProperty, Principal principal);

    Property createProperty(Property newProperty, Principal principal) ;

    int deletePropertyByID(int id, Principal principal);
}

File: client-side/src/services/PropertyService.js
import axios from 'axios';

export default{

    getProperties(){
        return axios.get('/api/property')
    },

    getProperty(id){
        return axios.get(`/api/property/${id}`)
    },

    createProperty(property){
        return axios.post('/api/property', property)
    },

    deleteProperty(id){
        return axios.delete(`/api/property/${id}`)
    },

    updateProperty(id, property){
        return axios.put(`/api/property/${id}`, property)
    }


}